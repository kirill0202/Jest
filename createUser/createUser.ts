type User = {
  name?: string;
  lastName?: string;
  age?: number;
  phone?: string;
  email?: string;
};
export const createUser = ({ name, lastName, age, phone, email }: User) => {
  return {
    phone,
    email,
    age,
    lastName,
    name,
  };
};
