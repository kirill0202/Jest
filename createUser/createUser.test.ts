import { createUser } from "./createUser";

describe("createUser", () => {
  it("should create user with firstname and lastname", () => {
    const user = createUser({ name: "Kirill", lastName: "Korobov" });
    expect(user).toMatchSnapshot();
  });

  it("should create user with firstname, lastname and phone", () => {
    const user = createUser({
      name: "Kirill",
      lastName: "Korobov",
      phone: "+7(123)456-78-90",
    });
    expect(user).toMatchSnapshot();
  });
  it("should create user with firstname, lastname, phone, email", () => {
    const user = createUser({
      name: "Kirill",
      lastName: "Korobov",
      phone: "+7(123)456-78-90",
      email: "test@example.com",
    });
    expect(user).toMatchSnapshot();
  });
});
