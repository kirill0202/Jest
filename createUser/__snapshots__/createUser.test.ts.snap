// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUser should create user with firstname and lastname 1`] = `
{
  "age": undefined,
  "email": undefined,
  "lastName": "Korobov",
  "name": "Kirill",
  "phone": undefined,
}
`;

exports[`createUser should create user with firstname, lastname and phone 1`] = `
{
  "age": undefined,
  "email": undefined,
  "lastName": "Korobov",
  "name": "Kirill",
  "phone": "+7(123)456-78-90",
}
`;

exports[`createUser should create user with firstname, lastname, phone, email 1`] = `
{
  "age": undefined,
  "email": "test@example.com",
  "lastName": "Korobov",
  "name": "Kirill",
  "phone": "+7(123)456-78-90",
}
`;
